mt_rng.cpp

```diff
@@ -0,0 +1,53 @@
+#include <ctime> // Include for time() function
+
+using namespace std;
+
+typedef unsigned long long ulong;
```

Comment:
I think we would prefer using uintxx_t from <stdint.h>, it makes it more readable.